### PR TITLE
site: solve display exception when use happy work theme in dark mode

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | solve display exception when use happy work theme in dark mode  |
| 🇨🇳 Chinese | 解决暗黑模式下使用快乐工作主题时首页展示异常问题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 70c696e</samp>

Support dark mode for antd theme in dumi docs. Modify `getAlgorithm` function in `.dumi/theme/layouts/GlobalLayout.tsx` to use dark algorithm if themes include 'dark'.

### <samp>🤖 Generated by Copilot at 70c696e</samp>

* Add dark mode support for the antd theme in the dumi documentation tool ([link](https://github.com/ant-design/ant-design/pull/44720/files?diff=unified&w=0#diff-5d7076c9ed6d07433cbe0e9ec9d4940ec94b53422e0475998ac2ce2e9caa3b46L49-R49),                             F

### CODE REVIEW

File: .dumi/theme/layouts/GlobalLayout.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ const getAlgorithm = (themes: ThemeName[] = []) =>
     if (theme === 'compact') {
       return antdTheme.compactAlgorithm;
     }
-    return antdTheme.defaultAlgorithm;
+    return themes.includes('dark') ? antdTheme.darkAlgorithm : antdTheme.defaultAlgorithm;
```
Comment (yoyo837) [inline .dumi/theme/layouts/GlobalLayout.tsx:49]: 前面的` if (theme === 'dark') {`覆盖不了?

Comment (yoyo837) [inline .dumi/theme/layouts/GlobalLayout.tsx:49]: 如果需要重新兜底, 是不是在前面 some一次好一些? includes 内部会重复循环吗? 蹲一个大神解答.

Comment (kiner-tang) [reply .dumi/theme/layouts/GlobalLayout.tsx:49]: 不行的，暗黑模式和快乐工作主题是可以同时存在的，url 类似：
http://localhost:8001/index-cn?theme=dark&theme=happy-work

Comment (kiner-tang) [reply .dumi/theme/layouts/GlobalLayout.tsx:49]: 其实还好，我们的 themes 说多了也就 2 ~ 3 个，我们其他地方也都是直接用 includes 判断的

Comment (zombieJ) [reply .dumi/theme/layouts/GlobalLayout.tsx:49]: 这个逻辑不太对，`theme` map 里不应该再依赖 `themes`。不如抽一下...

**Review by MadCcc — APPROVED** [review]

File: .dumi/theme/layouts/GlobalLayout.tsx [hunk 2]
```diff
@@ -39,15 +39,17 @@ const RESPONSIVE_MOBILE = 768;
...
+    .map((theme) => {
+      if (theme === 'dark') {
+        return antdTheme.darkAlgorithm;
+      }
+      if (theme === 'compact') {
+        return antdTheme.compactAlgorithm;
+      }
+      return null;
+    })
+    .filter((item) => item);
```
Comment (li-jia-nan) [inline .dumi/theme/layouts/GlobalLayout.tsx:52]: .filter(Boolean);

**Pull request merged**